### PR TITLE
1488 add delete vpc action

### PR DESCRIPTION
**Pull request by james-nesbitt**
- new registered action for delete vpc in c7n vpc resources

closes https://github.com/cloud-custodian/cloud-custodian/issues/1488

### CODE REVIEW

File: c7n/resources/vpc.py [hunk 1]
```diff
@@ -85,6 +85,39 @@ def process(self, resources):
...
+      .. code-block:: yaml
+
+        - name: aws-ec2-vpc-delete
+          resource: vpc
+          action:
+            - type: delete
+
+    """
+    schema = type_schema('delete')
+    permissions = ('ec2:DeleteVpc')
```
Comment (kapilt) [inline c7n/resources/vpc.py:105]: this needs a trailing comma after the string.

Comment (james-nesbitt) [reply c7n/resources/vpc.py:105]: will add that. There is an incoming change to fix the vpcid.

Comment (james-nesbitt) [reply c7n/resources/vpc.py:105]: thanks to @pgedara who sorted this out.

**Comment by james-nesbitt** [comment]
@kapilt all tests and codecov pass for this change (thanks to @pgedara.)
Technically, we are ready to merge.  There should be a discussion about whether or not this functionality should be merged (is it dangerous, or tricky enough that it would be difficult to support)

**Comment by james-nesbitt** [comment]
If desired, we can squash down to 1 commit

**Comment by kapilt** [comment]
I think with a good doc string (which is how the reference docs get made) and making clear that this is only intended and ever going to work for empty vpcs, I'm game.

I've got enough functional tests making vpcs, I can see the utility, but for the general case, something like aws-nuke is going to be more appropriate.

**Comment by kapilt** [comment]
actually we should rename this to `delete-empty` to be more explicit on its intent and functionality.

**Pull request merged**